react/admin/CostCenterAddressList.tsx

```diff
@@ -0,0 +1,239 @@
+        regex.test(item.postalCode) ||
+        regex.test(item.country) ||
+        (typeof item.receiverName === 'string' &&
+          regex.test(item.receiverName)) ||
+        regex.test(item.city) ||
+        regex.test(item.state) ||
+        regex.test(item.street) ||
+        (item.complement && regex.test(item.complement)) ||
+        (item.neighborhood && regex.test(item.neighborhood)) ||
+        (item.reference && regex.test(item.reference))
+      )
+    })
+  }
+
+  const handleRowsChange = (_: any, value: string) => {
+    const newTableLength = parseInt(value, 10)
+
+    setState({
+      ...state,
+      tableLength: newTableLength,
+      currentPage: 1,
+      currentItemFrom: 1,
+      currentItemTo: Math.min(newTableLength, state.filteredItems.length),
+      slicedData: state.filteredItems.slice(0, newTableLength),
+    })
+  }
+
+  const handleInputSearchSubmit = (e: { target: { value: string } }) => {
+    const searchValue = e?.target?.value
+    const filteredItems = filterItems(searchValue)
+
+    setState({
+      ...state,
+      searchValue,
+      filteredItems,
+      slicedData: filteredItems.slice(0, state.tableLength),
+      itemsLength: filteredItems.length,
+    })
+  }
+
+  const handleNextClick = () => {
+    const newPage = state.currentPage + 1
+    const itemFrom = state.currentItemTo + 1
+    const itemTo = Math.min(
+      itemFrom + state.tableLength - 1,
+      state.filteredItems.length
+    )
+
+    setState({
+      ...state,
+      currentPage: newPage,
+      currentItemFrom: itemFrom,
+      currentItemTo: itemTo,
+      slicedData: state.filteredItems.slice(itemFrom - 1, itemTo),
+    })
+  }
+
+  const handlePrevClick = () => {
+    if (state.currentPage === 1) return
+
+    const newPage = state.currentPage - 1
+    const itemFrom = state.currentItemFrom - state.tableLength
+    const itemTo = state.currentItemFrom - 1
+
+    setState({
+      ...state,
+      currentPage: newPage,
+      currentItemFrom: itemFrom,
+      currentItemTo: itemTo,
+      slicedData: state.filteredItems.slice(itemFrom - 1, itemTo),
+    })
+  }
+
+  const lineActions = [
+    {
+      label: () => `${formatMessage(messages.addressEdit)}`,
+      onClick: ({ rowData }: any) => {
+        handleEditAddressModal(rowData.addressId)
+      },
+    },
+    {
+      label: () => formatMessage(messages.addressDelete),
+      isDangerous: true,
+      onClick: ({ rowData }: any) => {
+        handleDeleteAddressModal(rowData.addressId)
+      },
+    },
+  ]
+
+  return (
+    <Table
+      fullWidth
+      schema={jsonschema}
+      items={state.slicedData}
+      emptyStateLabel={state.emptyStateLabel}
+      toolbar={{
+        inputSearch: {
+          value: state.searchValue,
+          placeholder: formatMessage(messages.searchAddress),
+          onChange: (e: any) =>
+            setState({ ...state, searchValue: e.target.value }),
+          onClear: () => {
+            const filteredItems = filterItems('')
```

Comment:
Maybe exctract this to a independent function to be called by `onClear` to improve readability